src/plugins/data/common/field_formats/field_format.ts

```diff
@@ -160,32 +168,31 @@ export abstract class FieldFormat {
   }
 
   static from(convertFn: FieldFormatConvertFunction) {
-    return createCustomFieldFormat(FieldFormat.toConvertObject(convertFn));
+    return createCustomFieldFormat(convertFn);
   }
 
   private static setupContentType(
     fieldFormat: IFieldFormat,
-    convert: FieldFormatConvert | FieldFormatConvertFunction
+    convert: Partial<FieldFormatConvert> | FieldFormatConvertFunction = {}
```

Comment:
unrelated to pr: do we actually use both of this options ? (passing in ConvertFunction or passing in the Convert object) ? have you maybe checked if we could change that and go only with one of the options ?